src/twig/linearize/Callable/src/grid.hpp

```diff
@@ -16,14 +16,19 @@ void grid( ForwardIterator& first,
            Midpoint&& midpoint ){
   auto xLeft = *first; ++first;
   auto yLeft = functor( xLeft );
+
+  decltype( xLeft ) xRight;
+  decltype( yLeft ) yRight;
   while ( first != last ){
-    auto xRight = *first; ++first;
-    auto yRight = functor( xRight );
+    xRight = *first; ++first;
+    yRight = functor( xRight );
     this->bin( xLeft, xRight, yLeft, yRight,
                std::forward<Functor>(functor),
                std::forward<Convergence>(criterion),
                std::forward<Midpoint>(midpoint) );
     xLeft = xRight;
     yLeft = yRight;
   }
+  this->xStack_->push_back( xRight );
+  this->yStack_->push_back( yRight );
```

Comment:
When first == last from the get-go, you will be pushing back default initialised values. Are you sure you want to do that?
Using
```cpp
auto xRight = xLeft;
auto yRight = yLeft;
```
may be a better idea to initialise these values.

Reply:
Yeah, I'm not doing this here anymore.

Reply:
Okay, upon further investigation this needs to stay here. Without these lines, the right most value of the input grid (and associated functor evaluation) does not get added. Without this, other codes downstream (i.e., RECONR) get errors because the arrays don't line up properly.

I'm not sure what to do about the situation where we begin with `first == last` right off the bat. Suggestions?